### ISSUE TITLE
[ENH] Discretizer transformer for post-processing

### ISSUE THREAD

**Issue by aiwalter** [alan-turing-institute/sktime#2443]
**Is your feature request related to a problem? Please describe.**
In real world, many business cases need a discrete forecast and no float numbers, but all of  `sktime` forecasters we have return float predictions.

**Describe the solution you'd like**
Implment a transformer `Discretizer` with a param to round up or round down. 

**Describe alternatives you've considered**

**Additional context**
Post-processing

**Comment by fkiraly** [comment]
sounds like a good first issue?

**Comment by fkiraly** [comment]
Should we also add a parameter "round to next multiple of", or a parameter that contains a finite set of values to round to?

**Comment by fkiraly** [comment]
Might also be worth allowing to encode intervals somehow, e.g., round to the next points contained in a `pd.IntervalIndex`

**Comment by AJarman** [comment]
Hi @fkiraly i'd like to pick this up please - it looks like it would be just implementing the `BaseTransformer._fit()` method?
My suggested implementation is
- `BaseTransformer`  child class
- _fit() method to round to integers (initial requirement for discrete values) as default.
- Constructor /`__init__` taking optional arguments : 
  - Round to dp `round_to_dp:Optional[int]=None`
  - Round to multiple of: `round_to_multiple:Optional[Union[float,int]]=None` 
  - List of finite values to round to `round_to_list:Optional[Union[float,int]]=None`
 
I am a little confused by the `pd.IntervalIndex` but we can clear that up later 👍

**Comment by fkiraly** [comment]
@AJarman, sure!

I'm not sure though why you want to implement the `_fit` method. Should the rounding not happen in the `_transform` method? There's nothing to fit as far as I can see.

Probably just a typo of yours? Otherwise your explanation looks good.

Re `pd.IntervalIndex`, just ignore that.

**Comment by AJarman** [comment]
Hello,
Yes I have started the implementation, I meant `_transform` of course 😄  that was a strange typo.

I am assuming that we don't want to store the old (untransformed/rounded) values and return them in `_inverse_transform`?
My understanding is that this would be the last step in a `TransformedTargetForecaster` pipeline and therefore the only method used is actually `_transform()`?

**Comment by fkiraly** [comment]
> I am assuming that we don't want to store the old (untransformed/rounded) values and return them in `_inverse_transform`?

That wouldn't work, since in general we'd be seeing new indices.

What I'd do is to have `inverse_transform` just to return `X`, since the rounded value itself is in a sense the "best guess" at what it was rounded from

**Comment by fkiraly** [comment]
> My understanding is that this would be the last step in a `TransformedTargetForecaster` pipeline and therefore the only method used is actually `_transform()`?

Yes, that is the usage intended by @aiwalter and probably typical usage, although we shouldn't rule out that users may want to use it elsewhere.

**Comment by aiwalter** [comment]
> What I'd do is to have inverse_transform just to return X, since the rounded value itself is in a sense the "best guess" at what it was rounded from

For this case, we have the tag `"skip-inverse-transform": True`, so no `inverse_transform` should be implemented. The method `inverse_transform` should only be implemented if its actually possible to inverse the transformation. As this is not the case here, we can leave it empty as e.g. also for `Imputer`

**Comment by fkiraly** [comment]
but isn´t the identity an "approximate inverse" here?

**Comment by aiwalter** [comment]
As a user, I expect from `inverse_transform` not an approximation

**Comment by AJarman** [comment]
Sorry I have been slow with this, unfortunately work has been busy.
Hopefully I can pick this up again soon. 
I am interesting in completing the holidays transformer as well.

**Comment by fkiraly** [comment]
@AJarman, great to hear back from you! Looking forward to the PR.
